bitcoin/src/amount.rs

```diff
@@ -499,7 +499,7 @@ impl Amount {
     pub const fn from_sat(satoshi: u64) -> Amount { Amount(satoshi) }
 
     /// Gets the number of satoshis in this [`Amount`].
-    pub fn to_sat(self) -> u64 { self.0 }
+    pub const fn to_sat(self) -> u64 { self.0 }
```

Comment:
Just an observation; in a way, this implicitly commits us to the internal representation used in `Amount` because we are committing to never using non-const code to get the sats equivalent of the inner representation. In this case I doubt we will ever use anything other than a `u64` but for other types this may be something to keep in mind?

Reply:
It commits us into never making representation so obscure that this would require a syscall (note: allocators also make syscalls sometimes). I can't imagine the drug that would make us so insane to do this for `Amount`.

I also don't see it as likely for other types. I've recently realized that there's no IO in `bitcoin` except for communicating with peers and storing data both of which we don't do so the only things that may become non-const are those that would start unconditionally allocating. For instance if a new address variant is added that has to allocate - but even then we can construct all other variants.

Reply:
Oh, crap, there's another reason: language limitations. They may make us unable to add some features.

Reply:
Yes, but that only means that we can't change the representation of `Amount` without changing the MSRV. So in some sense it's not a permanent limitation, and will inconvenience us at most.